apis/addons/v1alpha1/addons_types.go

```diff
@@ -5,18 +5,40 @@ import (
 )
 
 // AddonSpec defines the desired state of Addon.
-type AddonSpec struct{}
+type AddonSpec struct {
+	Namespaces []AddonNamespace `json:"namespaces,omitempty"`
+}
+
+type AddonNamespace struct {
+	Name string `json:"name"`
+}
+
+const (
+	// Available condition indicates that all resources for the Addon are reconciled and healthy
+	Available = "Available"
+)
 
 // AddonStatus defines the observed state of Addon
 type AddonStatus struct {
 	// The most recent generation observed by the controller.
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
 	// Conditions is a list of status conditions ths object is in.
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
-	// Human readable status aggregated from conditions.
-	Phase string `json:"phase,omitempty"`
+	// DEPRECATED: This field is not part of any API contract
+	// it will go away as soon as kubectl can print conditions!
+	// Human readable status - please use .Conditions from code
+	Phase AddonPhase `json:"phase,omitempty"`
 }
 
+type AddonPhase string
+
+// These are the valid phases of an Addon
+const (
+	Pending     AddonPhase = "Pending"
```

Comment:
Prefixing Phases with `Phase-` would make sure that there is no (or less) confusion between Condition Types like "Available" and Phases. This also prevents naming collisions with new ConditionTypes in the future WDYT?
tldr; ConditionType const will have no suffix or prefix, but Phases will always have a Prefix.

```suggestion
	PhasePending     AddonPhase = "Pending"
```

Reply:
Alrighty :) I see the value in this. :+1: 

I wouldn't have come up with this on my own because I try to rely on the language server to suggest all variables of type `AddonPhase` and I don't think the 'plan ahead' point is moot, becasue the  collisions can be easily refactored when they actually occur.